plugins/modules/proxmox_kvm.py

```diff
@@ -978,6 +996,13 @@ def create_vm(self, vmid, newid, node, name, memory, cpu, cores, sockets, update
                                       if 'storage' != k])
             kwargs['efidisk0'] = efidisk0_str
 
+        # Flatten tpmstate0 option to a string so that it's a string which is what Proxmoxer and the API expect
+        if 'tpmstate0' in kwargs:
+            kwargs['tpmstate0'] = '{storage}:1,version=v{version}'.format(
```

Comment:
This looks like both storage and version should be required values in `tmpstate0`.

Reply:
If the user specifies `tmpstate0: {}`, should the string `None:1,version=vNone` be passed as `tmpstate0`? I would expect `tmpstate0: {}` to be invalid, since both `storage` and `version` have to be specified (since `None` in the string isn't a good idea).

Reply:
Also the handling of `efidisk0` in the module looks not correct to me.

Reply:
I see. I'd be happy to spend some time fixing up the rest of this module, but that ought to come in a separate PR.